### ISSUE TITLE
Mixed `!.` and `?.` should be warned

### ISSUE THREAD

**Issue by falsandtru** [microsoft/TypeScript#35071]
It is an anti-pattern. And must not do quick fix so.

**TypeScript Version:**  3.7.x-dev.20191105

**Search Terms:**

**Code**

```ts
document.querySelector('_')!.textContent!.split('') // old code
document.querySelector('_')?.textContent!.split('') ?? 0 // string[]
document.querySelector('_')?.textContent?.split('') ?? 0 // string[] | 0
```

**Expected behavior:**

```ts
document.querySelector('_')!.textContent!.split('') // old code
document.querySelector('_')?.textContent!.split('') ?? 0 // unsafe, warning
document.querySelector('_')?.textContent?.split('') ?? 0 // safe
```

**Actual behavior:**

```ts
document.querySelector('_')!.textContent!.split('') // old code
document.querySelector('_')?.textContent!.split('') ?? 0 // unsafe, no warning
document.querySelector('_')?.textContent?.split('') ?? 0 // safe
```

**Playground Link:** 

**Related Issues:**

**Comment by falsandtru** [comment]
If the compiler control details, the compiler would be possible to allow `?.` after `!.`.

**Comment by kitsonk** [comment]
Isn't this a just linting rule?  Whenever you use `!` it is potentially unsafe.  The compiler never warns, an error is an error.

**Comment by falsandtru** [comment]
When quick fix observe this rule, it is wasteful splitting.

**Comment by RyanCavanaugh** [comment]
@falsandtru some discussion about what exactly is different between this and #35025 would be nice

**Comment by falsandtru** [comment]
This issue has meaning without adopting #35025 and meaningless if #35025 is adopted. So this issue and #35025 are in opposition. This is the definitive difference.
